### PR TITLE
fix: check not stoppped instead of is alive

### PR DESCRIPTION
**Pull request by emord**
https://www.notion.so/peregrine-io/Failing-extractor-doesn-t-show-up-as-a-failed-run-in-ui-59c53f0f7a5b4da8bf4e4438a16cbb71
https://github.com/peregrine-io/peregrine/pull/3285

first commit is just running black for my sanity. upstream did this and i was trying to diff it. (upstream is dead so not bothering pushing to them atm)

second commit:

no obvious change from the change log but between python 3.7 and 3.8. ThreadPoolExecutor changed behavior such that submitting a job is actually asynchronous and we aren't guaranteed that the job has started when we submit it. the  bonobo "alive" check was just seeing if any of the nodes had been started. an error occurs whenever a bonobo node has started and finished but the next node hasn't even started yet.

i wanted to change the .alive property, but that actually fails more tests that test the flow of execution state. i'm less gung ho on changing that at the moment.

### CODE REVIEW

**Comment by emord** [comment]
3.10 failing is a dependency issue. not going to fix that since we're running 3.9 atm

File: bonobo/execution/strategies/executor.py [hunk 1]
```diff
@@ -35,11 +38,11 @@ def execute(self, graph, **kwargs):
                     exc_info=sys.exc_info(),
                 )
 
-            while context.alive:
+            while not context.stopped:
```
Comment (danaford) [inline bonobo/execution/strategies/executor.py:41]: 👀

**Review by danaford — APPROVED** [review]
This is amazing. So to ask the obvious question I assume calling `context.stopped` does not error even when a node hasn't started yet (though it has been submitted)?

**Comment by 1calorie** [comment]
nice digging

**Review by benrudolph — APPROVED** [review]
damn beastly. this is not an easy library to grok

**Comment by emord** [comment]
> I assume calling context.stopped does not error even when a node hasn't started yet (though it has been submitted)?

this context is the graph context. stopped, alive, and started all just check some instance variables on the node context. all of the node contexts are instantiated before submitting to the executor. 

> this is not an easy library to grok

it gets a lot easier with debug logging and type annotations

File: bonobo/execution/strategies/executor.py (hunk 1 above)
Comment (herg) [reply bonobo/execution/strategies/executor.py:41]: so obvious now that i see it

**Pull request merged**